packages/ses/src/module-static-record.js

```diff
@@ -0,0 +1,44 @@
+import * as babel from '@agoric/babel-standalone';
+// We are using the above import form, and referring to its default export
+// explicitly below like babel.default, because the proper construct causes a
+// Rollup error.
+// This form:
+//   import babel from '@agoric/babel-standalone';
+// And this variant:
+//   import { default as babel } from '@agoric/babel-standalone';
+// Both produce:
+//   Error: 'default' is not exported by .../@agoric/babel-standalone/babel.js
+import { makeModuleAnalyzer } from '@agoric/transform-module';
+
+const analyze = makeModuleAnalyzer(babel.default);
+
+// moduleAnalyses are the private data of a ModuleStaticRecord.
+// We use moduleAnalyses in the loader/linker to look up
+// the analysis corresponding to any ModuleStaticRecord constructed by an
+// importHook.
+// This WeakMap is shared among modules in this package, but not exposed to the
+// public.
+export const moduleAnalyses = new WeakMap();
+
+// ModuleStaticRecord captures the effort of parsing and analyzing module text
+// so a cache of ModuleStaticRecords may be shared by multiple Compartments.
+export class ModuleStaticRecord {
+  constructor(text) {
+    const analysis = analyze({ string: text });
+    moduleAnalyses.set(this, analysis);
+
+    this.imports = Object.keys(analysis.imports).sort();
+
+    Object.freeze(this);
+    Object.freeze(this.imports);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  toString() {
```

Comment:
I suspect it is more conventional to override the `Symbol.toStringTag` on `ModuleStaticRecord.prototype`, but I don't know. Unfortunately, that would need to be done by assignment before exporting the class.

How do we handle the stringification of other shimmed class-like abstractions?

Reply:
This is a pattern I copied from the Compartment constructor.

Reply:
Cool. Resolving.